.github/workflows/build.yml

```diff
@@ -19,17 +20,147 @@ jobs:
+          toolchain: nightly-2021-05-11
+          override: true
+          target: wasm32-unknown-unknown
+    - name: Install System Level Dependencies
+      run: sudo apt-get install build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev git cmake perl pkg-config curl llvm-10 clang-10 libclang-10-dev
+    - name: Download Intel SGX SDK
+      run: wget https://download.01.org/intel-sgx/sgx-linux/2.12/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Install Intel SGX SDK
+      run: chmod +x ./sgx_linux_x64_sdk_2.12.100.3.bin && echo -e 'no\n/opt/intel' | sudo ./sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v2.4.0
+      with:
+        node-version: 14.x
+    - name: Install yarn 2
+      run: sudo npm install -g yarn && yarn set version berry
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+    
+    - name: Build pRuntime
+      run: source /opt/intel/sgxsdk/environment && cd ./standalone/pruntime/ && SGX_MODE=SW make
+    - name: Save pruntime binaries [./bin/app]
+      uses: actions/upload-artifact@v2
+      with:
+        name: pruntime-binaries-app
+        path: ./standalone/pruntime/bin/app
+    - name: Save pruntime binaries [./bin/enclave.signed.so]
+      uses: actions/upload-artifact@v2
+      with:
+        name: pruntime-binaries-enclave
+        path: ./standalone/pruntime/bin/enclave.signed.so
+    - name: Save pruntime binaries [./bin/Rocket.toml]
+      uses: actions/upload-artifact@v2
+      with:
+        name: pruntime-binaries-rocket
+        path: ./standalone/pruntime/bin/Rocket.toml
+  
+  e2e-test:
+    name: e2e-test
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Install latest nightly
+      uses: actions-rs/toolchain@v1
+      with:
+          toolchain: nightly-2021-05-11
+          override: true
+          target: wasm32-unknown-unknown
+    - name: Install System Level Dependencies
+      run: sudo apt-get install build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev git cmake perl pkg-config curl llvm-10 clang-10 libclang-10-dev
+    - name: Download Intel SGX SDK
+      run: wget https://download.01.org/intel-sgx/sgx-linux/2.12/distro/ubuntu20.04-server/sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Install Intel SGX SDK
+      run: chmod +x ./sgx_linux_x64_sdk_2.12.100.3.bin && echo -e 'no\n/opt/intel' | sudo ./sgx_linux_x64_sdk_2.12.100.3.bin
+    - name: Setup Node.js environment
+      uses: actions/setup-node@v2.4.0
+      with:
+        node-version: 14.x
+    - name: Install yarn 2
+      run: sudo npm install -g yarn && yarn set version berry
+    - uses: actions/checkout@v2
+      with:
+        submodules: 'true'
+    
+    - name: Remove folders
+      run: rm -rf ./target/release/phala-node ./target/release/pherry ./standalone/pruntime/bin/app ./standalone/pruntime/bin/enclave.signed.so ./standalone/pruntime/bin/Rocket.toml
```

Comment:
Why a removing action here?  They should not exists in a fresh checkout.

Reply:
Done.